deps/rabbit/src/rabbit_binding.erl

```diff
@@ -100,13 +100,17 @@ recover(XNames, QNames) ->
 
 recover_semi_durable_route(Gatherer, Binding, Src, Dst, ToRecover, Fun) ->
     case sets:is_element(Dst, ToRecover) of
-        true  -> {ok, X} = rabbit_exchange:lookup(Src),
-                 ok = gatherer:fork(Gatherer),
-                 ok = worker_pool:submit_async(
-                        fun () ->
-                                Fun(Binding, X),
-                                gatherer:finish(Gatherer)
-                        end);
+        true  ->
+            case rabbit_exchange:lookup(Src) of
+                {ok, X} ->
+                    ok = gatherer:fork(Gatherer),
+                    ok = worker_pool:submit_async(
+                           fun () ->
+                                   Fun(Binding, X),
+                                   gatherer:finish(Gatherer)
+                           end);
+                {error, not_found} -> ok
```

Comment:
Can you please add a warning here? If we have gone through this code path, something has gone wrong